client_test.go

```diff
@@ -21,24 +21,20 @@ func TestClientSync(t *testing.T) {
 	c := new(Client)
 	r, _, e := c.Exchange(m, addrstr)
 	if e != nil {
-		t.Logf("failed to exchange: %s", e.Error())
-		t.Fail()
+		t.Errorf("failed to exchange: %s", e.Error())
 	}
 	if r != nil && r.Rcode != RcodeSuccess {
-		t.Log("failed to get an valid answer")
-		t.Fail()
-		t.Logf("%v\n", r)
+		t.Error("failed to get an valid answer")
+		t.Errorf("%v\n", r)
```

Comment:
There are a lot of cases where log lines could be combined or restructured.  I opted to leave that out of scope for this round of changes.